addons/mail/models/mail_thread.py

```diff
@@ -694,6 +694,10 @@ def _notify_get_action_link(self, link_type, **kwargs):
             params['token'] = token
 
         link = '%s?%s' % (base_link, url_encode(params))
+
+        if hasattr(self, 'get_base_url'):
+            link = self[0].get_base_url() + link
```

Comment:
`[0]` will crash if self is empty recordset

Reply:
Good catch, done